tests/unit/s2n_handshake_io_async_test.c

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include <s2n.h>
+
+#include "s2n_test.h"
+#include "testlib/s2n_testlib.h"
+
+#include "utils/s2n_result.h"
+
+/* Get access to s2n_handshake_read_io */
+#include "tls/s2n_handshake_io.c"
+
+bool async_blocked = false;
+size_t blocking_handler_count = 0;
+static int s2n_blocking_handler(struct s2n_connection *conn)
+{
```

Comment:
We could use `S2N_RESULT` for the static function

Reply:
We can't, it has to have a specific signature to be used as a message handler in the state machine.